util/topsql/topsql.go

```diff
@@ -50,17 +50,19 @@ func SetupTopSQL() {
 	singleTargetDataSink = reporter.NewSingleTargetDataSink(remoteReporter)
 
 	globalTopSQLReport = remoteReporter
-
-	tracecpu.GlobalSQLCPUProfiler.SetCollector(remoteReporter)
-	tracecpu.GlobalSQLCPUProfiler.Run()
 	stmtstats.RegisterCollector(remoteReporter)
 	stmtstats.SetupAggregator()
 }
 
+// SetupTopSQLForTest sets up the global top-sql reporter, it's exporting for test.
+func SetupTopSQLForTest(r reporter.TopSQLReporter) {
+	globalTopSQLReport = r
+}
+
 // RegisterPubSubServer registers TopSQLPubSubService to the given gRPC server.
 func RegisterPubSubServer(s *grpc.Server) {
-	if globalTopSQLReport != nil {
-		service := reporter.NewTopSQLPubSubService(globalTopSQLReport)
+	if register, ok := globalTopSQLReport.(reporter.DataSinkRegisterer); ok {
```

Comment:
Why not simply keep `globalTopSQLReport` in type `reporter.RemoteTopSQLReporter`? Then you don't need a type check any more.

Reply:
`globalTopSQLReport` use `reporter.TopSQLReporter` is for the test, since some the use the `mock.TopSQLCollector`